### DISCUSSION TITLE
Best practices for adding non-pfsense related logs?

### DISCUSSION THREAD

**Discussion by ightnapovial** [pfelk/pfelk#299] [Q&A]
I'm trying to add pihole logs to the pfelk server using [elk-hole](https://github.com/nin9s/elk-hole) and am finding that the configs from the two projects seem to cause issues with one another. For example, on my working pfelk install if I add the logstash configs from elk-hole and set the logs to come in on a new port, I see logs coming into the stack, however 100% are tagged with grok parse errors. I've confirmed that the elk-hole project configs work as intended on their own on a fresh stack without pfelk.

I'm relatively new to elastic so I'm fairly confident that I'm screwing something up myself. I'm just curious if this is the right approach to combining the two projects or if I should go another route. Anyone have any tips that might put me on the right path?

**Comment by ightnapovial** [comment]
I tracked down the issue I was having with grok parsing errors, 45-cleanup.conf changes the "message" field to "[event][original]" and elf-hole was expecting "message." Flipping to event.original in the elk-hole conf file got the logs parsed correctly.

I am still curious if this is the right way to go or not if any elastic SME's have any input.

**Reply by a3ilson** [reply]
@ightnapovial - in addition to the port, you also need to specify within the pipelines.yml - example below:

```
# pipelines.yml
################################################################################
# Version: 21.03                                                               #
# Required: Yes                                                                #
# Description: This is a required file for a pfelk installation                #
# This file is where you define your pipelines. You can define multiple.       #
# For more information on multiple pipelines, see the documentation:           #
#   https://www.elastic.co/guide/en/logstash/current/multiple-pipelines.html   #
#                                                                              #
################################################################################
#
- pipeline.id: pfelk
  path.config: "/etc/pfelk/conf.d/*.conf"

 - pipeline.id: elk-hole
  path.config: "/etc/elk-hole/*.conf"
```

With the above, you would need to copy your elk-hole *.conf files to `/etc/elk-hole/`
